src/modules/jcms_rest/tests/src/Functional/EndpointValidatorTest.php

```diff
@@ -86,17 +92,25 @@ public function dataProvider() : array {
    *   The HTTP method/verb to be used.
    * @param string $endpoint
    *   The endpoint/URI to test.
-   * @param string $mime_type
-   *   The accept header mime type to send.
+   * @param string $media_type
+   *   The accept header media type to send.
    */
-  public function testValidateEndpointsAgainstRaml(string $method, string $endpoint, string $mime_type) {
+  public function testValidateEndpointsAgainstRaml(string $method, string $endpoint, string $media_type) {
     $request = new Request($method, $endpoint, [
-      'Accept' => $mime_type,
+      'Accept' => $media_type,
     ]);
     $response = $this->client->send($request);
     $json_decoder = new JsonDecoder();
     $messageValidator = new FakeHttpsMessageValidator(new JsonMessageValidator(new PuliSchemaFinder($this->resourceRepository), $json_decoder), $json_decoder);
     $messageValidator->validate($response);
   }
 
+  public function testNotExistent() {
+    $request = new Request('GET', '/subjects/does-not-exist', [
+      'Accept' => 'application/vnd.elife.subject+json; version=1',
+    ]);
+    $response = $this->client->send($request);
+    $this->assertEquals(404, $response->getStatusCode());
```

Comment:
now you could check also the body to be a JSON document with the `title` property